Observer/Model/ProcessOrder/CardsMetadata.php

```diff
@@ -67,9 +76,11 @@ public function execute(Observer $observer)
                 'acquirerResponseMessage'
             ];
 
+            $payment = $order->getPayment();
             foreach ($keys as $key) {
-                $this->populateCardData($data, $payment, $key);
+                $this->populateCardData($data, $paymentData, $key, $payment);
             }
+            $this->paymentResource->save($payment);
```

Comment:
Same as previous comment, this one is expecting AbstractModel so if the implementation is not Payment it might fail.